mitiq/ddd/rules/rules.py

```diff
@@ -16,3 +16,169 @@
+            in the given slack window.
+            E.g. given slack_length = 20 the spacing defaults to 6 and this
+            rule returns the sequence:
+            ──I──I──I──I──I──I──X──I──I──I──I──I──I──X──I──I──I──I──I──I──
+    Returns:
+        An XX digital dynamical decoupling sequence, as a Cirq circuit.
+    """
+    xx_sequence = general_rule(
+        slack_length=slack_length,
+        spacing=spacing,
+        gates=[X, X],
+    )
+    return xx_sequence
+
+
+def xyxy(slack_length: int, spacing: int = -1) -> Circuit:
+    """Returns an XYXY digital dynamical decoupling sequence, based on inputs.
+
+    Args:
+        slack_length: Length of idle window to fill.
+        spacing: How many identity spacing gates to apply between dynamical
+            decoupling gates, as a non-negative int. Negative int corresponds
+            to default. Defaults to maximal spacing that fits a single sequence
+            in the given slack window.
+            E.g. given slack_length = 20 the spacing defaults to 3 and this
+            rule returns the sequence:
+            ──I──I──I──I──X──I──I──I──Y──I──I──I──X──I──I──I──Y──I──I──I──
+    Returns:
+        An XYXY digital dynamical decoupling sequence, as a Cirq circuit.
+    """
+    xyxy_sequence = general_rule(
+        slack_length=slack_length,
+        spacing=spacing,
+        gates=[X, Y, X, Y],
+    )
+    return xyxy_sequence
+
+
+def yy(slack_length: int, spacing: int = -1) -> Circuit:
+    """Returns a YY digital dynamical decoupling sequence, based on inputs.
+
+    Args:
+        slack_length: Length of idle window to fill.
+        spacing: How many identity spacing gates to apply between dynamical
+            decoupling gates, as a non-negative int. Negative int corresponds
+            to default. Defaults to maximal spacing that fits a single sequence
+            in the given slack window.
+            E.g. given slack_length = 20 the spacing defaults to 6 and
+            this rule returns the sequence:
+            ──I──I──I──I──I──I──Y──I──I──I──I──I──I──Y──I──I──I──I──I──I──
+    Returns:
+        An YY digital dynamical decoupling sequence, as a Cirq circuit.
+    """
+    yy_sequence = general_rule(
+        slack_length=slack_length,
+        spacing=spacing,
+        gates=[Y, Y],
+    )
+    return yy_sequence
+
+
+def repeated_rule(slack_length: int, gates: List[Gate]) -> Circuit:
+    """Returns a general digital dynamical decoupling sequence that repeats
+    until the slack is filled without spacing, up to a complete repetition.
+
+    Args:
+        slack_length: Length of idle window to fill.
+        gates: A list of single qubit Cirq gates to build the rule. E.g. [X, X]
+            is the xx sequence, [X, Y, X, Y] is the xyxy sequence.
+    Returns:
+        A repeated digital dynamical decoupling sequence, as a Cirq circuit.
+    Note:
+        Where the general rule requires pre-knowledge of the amount of slack,
```

Comment:
```suggestion
        Where :fun:`.general_rule()` fills a slack window with single sequence,
```

I know what you mean but in practice both functions need to know `slack_length` as input.